src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs

```diff
@@ -17,44 +18,65 @@
 
 namespace Microsoft.Azure.WebJobs.Script
 {
-    public class WorkerFunctionMetadataProvider : IFunctionMetadataProvider
+    public class AggregateFunctionMetadataProvider : IFunctionMetadataProvider
     {
         private readonly Dictionary<string, ICollection<string>> _functionErrors = new Dictionary<string, ICollection<string>>();
         private readonly ILogger _logger;
         private readonly IFunctionInvocationDispatcher _dispatcher;
         private ImmutableArray<FunctionMetadata> _functions;
+        private IFunctionMetadataProvider _hostFunctionMetadataProvider;
 
-        public WorkerFunctionMetadataProvider(ILogger<WorkerFunctionMetadataProvider> logger, IFunctionInvocationDispatcher invocationDispatcher)
+        public AggregateFunctionMetadataProvider(
+            ILogger<AggregateFunctionMetadataProvider> logger,
+            IFunctionInvocationDispatcher invocationDispatcher,
+            IFunctionMetadataProvider hostFunctionMetadataProvider)
         {
             _logger = logger;
-            _dispatcher = invocationDispatcher ?? throw new ArgumentNullException(nameof(invocationDispatcher));
```

Comment:
why move this inside the function?

Reply:
Discussed offline. _dispatcher can be null if using hostIndexing and it cannot be null if using workerIndexing.